### PR TITLE
EVG-2704 Agent should not update during a task group

### PR DESCRIPTION
**Pull request by bsamek**

### CODE REVIEW

File: agent/agent.go [hunk 1]
```diff
@@ -97,21 +97,27 @@ func (a *Agent) loop(ctx context.Context) error {
...
 			if nextTask.TaskId != "" {
 				if nextTask.TaskSecret == "" {
 					return errors.New("task response missing secret")
 				}
-				tc = a.prepareNextTask(ctx, nextTask, &tc)
+				tc, exit = a.prepareNextTask(ctx, nextTask, &tc)
+				if exit {
+					// Query for next task, this time with an empty task group,
+					// to get a ShouldExit from the API, and set NeedsNewAgent.
+					continue
```
Comment (tychoish) [inline agent/agent.go]: use loop levels. this continue just continues the select statement not the for loop, which might be the same, but could be confusing in the future.

File: taskrunner/error_handler.go [hunk 2]
```diff
@@ -107,8 +107,7 @@ func processErrorItem(rec hostRecord, errors errorRecord) string {
 		lines = append(lines, "Action: Disabled Host")
 		err := errors.host.DisablePoisonedHost()
 
-		job := units.NewDecoHostNotifyJob(env, errors.host, err,
-			"host encountered consecutive set up failures")
+		job := units.NewDecoHostNotifyJob(env, errors.host, err)
```
Comment (tychoish) [inline taskrunner/error_handler.go]: I would revert this change and the change to units.

File: agent/agent.go (hunk 1 above)
Comment (bsamek) [reply agent/agent.go]: fixed

**Comment by bsamek** [comment]
I also unset the running task if the agent is exiting.

File: model/host/host.go [hunk 3]
```diff
@@ -363,6 +363,23 @@ func (host *Host) ClearRunningTask(prevTaskId string, finishTime time.Time) erro
 	return nil
 }
 
+// UnsetRunningTask unsets the running task fields. It does NOT set last task completed fields.
+func (host *Host) UnsetRunningTask() error {
```
Comment (tychoish) [inline model/host/host.go]: it's probably right to also update last communication time here?
